### ISSUE TITLE
ImmutableRecordLogic calls fromInt instead of fromFloat

### ISSUE THREAD

**Issue by martin-schilling** [proophsoftware/event-machine#94]
When the fraction points of the float inside a float VO happen to be zero it gets converted to an integer inside the JSON representation eg. `["prop1" => 3.2, "prop2" => 4.0]` will result in `{"prop1":3.2,"prop2":4}`. When event machine now tries to replay the events to recreate the aggregate state it will encounter an error because it is going to call `::fromInt()` for `prop2` which will obviously fail since `::fromFloat()` is the method that should be called instead. A simple solution might be to add a check like this here https://github.com/proophsoftware/event-machine/blob/master/src/Data/ImmutableRecordLogic.php#L295
```
return method_exists($type, 'fromInt')
    ? $type::fromInt($value)
    : $type::fromFloat($value);
```
What do you think @codeliner ? Is there a better solution?

**Comment by codeliner** [comment]
@martin-schilling good catch!
I'm fine with the solution. Can you provide a PR?

**Comment by codeliner** [comment]
@martin-schilling If you want to run the tests, here is an update how to do that after latest changes: https://github.com/proophsoftware/event-machine/commit/bfb8f300d7adff2e714d619221f8c14e32accfe2

**Comment by martin-schilling** [comment]
Yes I will do so tomorrow

**Comment by codeliner** [comment]
Released with [v0.16](https://github.com/proophsoftware/event-machine/releases/tag/v0.16.0)
